configure.ac

```diff
@@ -439,6 +439,7 @@ if test "x$enable_werror" = "xyes"; then
                         [AC_LANG_SOURCE([[struct A { virtual void f(); }; struct B : A { void f() final; };]])])
   AX_CHECK_COMPILE_FLAG([-Werror=unreachable-code-loop-increment],[ERROR_CXXFLAGS="$ERROR_CXXFLAGS -Werror=unreachable-code-loop-increment"],,[[$CXXFLAG_WERROR]])
   AX_CHECK_COMPILE_FLAG([-Werror=mismatched-tags], [ERROR_CXXFLAGS="$ERROR_CXXFLAGS -Werror=mismatched-tags"], [], [$CXXFLAG_WERROR])
+  AX_CHECK_COMPILE_FLAG([-Werror=implicit-fallthrough], [ERROR_CXXFLAGS="$ERROR_CXXFLAGS -Werror=implicit-fallthrough"], [], [$CXXFLAG_WERROR])
 fi
 
 if test "x$CXXFLAGS_overridden" = "xno"; then
```

Comment:
any reason to not enable the warning as well?

Reply:
You are right!

From https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html:
> `-Wimplicit-fallthrough` is the same as `-Wimplicit-fallthrough=3`...
>
> The `-Wimplicit-fallthrough=3` warning is enabled by `-Wextra`.

But that is _not_ the case for clang. Going to fix it now.

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/21430#issuecomment-799583722).